flutter_appauth/macos/Classes/AppAuthMacOSAuthorization.m

```diff
@@ -3,13 +3,21 @@
 @implementation AppAuthMacOSAuthorization
 
 - (id<OIDExternalUserAgentSession>)performAuthorization:(OIDServiceConfiguration *)serviceConfiguration clientId:(NSString*)clientId clientSecret:(NSString*)clientSecret scopes:(NSArray *)scopes redirectUrl:(NSString*)redirectUrl additionalParameters:(NSDictionary *)additionalParameters preferEphemeralSession:(BOOL)preferEphemeralSession result:(FlutterResult)result exchangeCode:(BOOL)exchangeCode {
+    NSString *codeVerifier = [OIDAuthorizationRequest generateCodeVerifier];
+    NSString *codeChallenge = [OIDAuthorizationRequest codeChallengeS256ForVerifier:codeVerifier];
+
     OIDAuthorizationRequest *request =
     [[OIDAuthorizationRequest alloc] initWithConfiguration:serviceConfiguration
                                                   clientId:clientId
                                               clientSecret:clientSecret
                                                     scopes:scopes
                                                redirectURL:[NSURL URLWithString:redirectUrl]
                                               responseType:OIDResponseTypeCode
+                                                     state:[OIDAuthorizationRequest generateState]
+                                                     nonce: nonce != nil ? nonce : [OIDAuthorizationRequest generateState]
```

Comment:
macOS build is failing due to a compilation error around here. could you take a look?